src/_tkagg.cpp

```diff
@@ -232,6 +264,90 @@ static PyMethodDef functions[] = {
     { NULL, NULL } /* sentinel */
 };
 
+#ifdef DYNAMIC_TKINTER
+// Functions to fill global TCL / Tk function pointers from tkinter module.
+
+#include <dlfcn.h>
+
+#if PY3K
+#define TKINTER_PKG "tkinter"
+#define TKINTER_MOD "_tkinter"
+// From module __file__ attribute to char *string for dlopen.
+#define FNAME2CHAR(s) (PyBytes_AsString(PyUnicode_EncodeFSDefault(s)))
+#else
+#define TKINTER_PKG "Tkinter"
+#define TKINTER_MOD "tkinter"
+// From module __file__ attribute to char *string for dlopen
+#define FNAME2CHAR(s) (PyString_AsString(s))
+#endif
+
+void *_dfunc(void *lib_handle, const char *func_name)
+{
+    // Load function, unless there has been a previous error.  If so, then
+    // return NULL.  If there is an error loading the function, return NULL
+    // and set error flag.
+    static int have_error = 0;
```

Comment:
This is fine, but is a sort of unidiomatic pattern.  Rather than using a static variable `has_error` here, why not just return NULL if there was an error and then call it differently from `_func_loader`, e.g.:

```
if (!((TCL_CREATE_COMMAND = (tcl_cc) _dfunc(tkinter_lib, "Tcl_CreateCommand")) &&
      (TCL_APPEND_RESULT = (tcl_cc) _dfunc(tkinter_lib, "Tcl_AppendResult")) &&
    ... etc ..
```